### PR TITLE
Add Sleep Feature

### PR DESCRIPTION
**Pull request by a-wild-chocolate**
Add Sleep feature, including updating UI, LifeTraker, Praser and Test

### CODE REVIEW

**Review by paturikarthik — APPROVED** [review]
Nice work! Everything looks good, and the organisation of the code is also consistent with previous iterations.

File: src/main/java/seedu/lifetrack/system/parser/ParserSleep.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            for (String part : parts) {
+                if (part.startsWith("t/")) {
+                    duration = Double.parseDouble(part.substring(2));
+                    if (duration < 0) {
+                        throw new InvalidInputException(getIncorrectSleepInputMessage());
+                    }
+                } else if (part.startsWith("d/")) {
+                    date = part.substring(2);
+                    if (!date.matches("\\d{6}")) {
+                        throw new InvalidInputException(getIncorrectSleepDateInputMessage());
```
Comment (owx0130) [inline src/main/java/seedu/lifetrack/system/parser/ParserSleep.java:24]: can we change the command format to `sleep add DURATION d/DDMMYY`, so the only difference is there is no t/ keyword. mostly for standardization across the 2 other features

**Review by owx0130 — COMMENTED** [review]
great job overall

File: src/main/java/seedu/lifetrack/system/parser/ParserSleep.java (hunk 1 above)
Comment (a-wild-chocolate) [reply src/main/java/seedu/lifetrack/system/parser/ParserSleep.java]: done

**Pull request merged**